framework/src/modules/chain/blocks/blocks.js

```diff
@@ -594,6 +716,130 @@ class Blocks extends EventEmitter {
 			}
 		);
 	}
+
+	/**
+	 * Block IDs are the same ~ Blocks are equal
+	 * @param block
+	 * @returns {*}
+	 * @private
+	 */
+	// eslint-disable-next-line class-methods-use-this
+	_handleSameBlockReceived(block) {
+		this.logger.debug('Block already processed', block.id);
+	}
+
+	/**
+	 * Block received is correct
+	 * @param block
+	 * @returns {Promise}
+	 * @private
+	 */
+	async _handleValidBlock(block) {
+		return this._processReceivedBlock(block);
+	}
+
+	/**
+	 * Double forging. Last block stands
+	 * @param block
+	 * @param lastBlock
+	 * @returns {*}
+	 * @private
+	 */
+	// eslint-disable-next-line class-methods-use-this
+	_handleDoubleForging(block, lastBlock) {
+		this.logger.debug(
+			'Delegate forging on multiple nodes',
+			block.generatorPublicKey
+		);
+		this.logger.debug(
+			`Last block ${lastBlock.id} stands, new block ${block.id} is discarded`
+		);
+		// TODO: Implement Proof of Misbehavior
+	}
+
+	/**
+	 * Tie break: two competing blocks by different delegates at the same height.
+	 * @param lastBlock
+	 * @param newBlock
+	 * @returns {Promise}
+	 * @private
+	 */
+	async _handleDoubleForgingTieBreak(newBlock, lastBlock) {
+		const block = cloneDeep(newBlock);
+		// It mutates the argument
+		const check = await this.blocksVerify.normalizeAndVerify(
+			block,
+			lastBlock,
+			this._lastNBlockIds
+		);
+
+		if (!check.verified) {
+			const errors = convertErrorsToString(check.errors);
+
+			this.logger.error(
```

Comment:
Ah sorry I meant
```
this.logger.error( { errors, blockId: block.id }, 'Fork Choice Case 4 recovery failed in verification and normalization');
```

Reply:
I was checking Bunyan docs and you can do this.logger.error(error, message), the first argument being specifically for error objects

Reply:
yes you can, also i thought having meta data in object is probably better than message because its easier to filter

Reply:
@shuse2 please see #3818